prover/prover_fri_utils/src/metrics.rs

```diff
@@ -0,0 +1,41 @@
+use std::time::Duration;
+use vise::{Buckets, EncodeLabelSet, EncodeLabelValue, Family, Histogram, Metrics};
+use zksync_types::proofs::AggregationRound;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelSet)]
+pub(crate) struct CircuitLabels {
+    pub circuit_type: u8,
+    pub aggregation_round: StageLabel,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelValue, EncodeLabelSet)]
+#[metrics(label = "stage", format = "wit_gen_{}")]
+pub(crate) struct StageLabel(AggregationRound);
+
+impl From<AggregationRound> for StageLabel {
+    fn from(round: AggregationRound) -> Self {
+        Self(round)
+    }
+}
+
+impl std::fmt::Display for StageLabel {
+    fn fmt(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        formatter.write_str(match self.0 {
+            AggregationRound::BasicCircuits => "basic_circuits",
+            AggregationRound::LeafAggregation => "leaf_aggregation",
+            AggregationRound::NodeAggregation => "node_aggregation",
+            AggregationRound::Scheduler => "scheduler",
+        })
+    }
```

Comment:
I'm surprised we didn't have this mapping. I checked - and indeed we didn't! @EmilLuta just wanted to draw your attention here - so that next time we reuse this mapping and not create new